packages/admin/src/EditDialog.tsx

```diff
@@ -85,10 +86,15 @@ const EditDialogInner: React.FunctionComponent<IProps & IHookProps> = ({ selecti
     let dirtyHandlerApi: IDirtyHandlerApi | undefined;
     const handleSaveClick = () => {
         if (dirtyHandlerApi) {
-            dirtyHandlerApi.submitBindings().then(() => {
-                setTimeout(() => {
-                    selectionApi.handleDeselect();
-                });
+            dirtyHandlerApi.submitBindings().then((values: Array<undefined | SubmissionErrors>) => {
+                // for final-form undefined means success, an obj means error
+                const failed = values.reduce((accumulator, value) => accumulator || value !== undefined, false);
```

Comment:
EditDialog should not just support forms (although in most cases there will be a form)

I think we should either:
- create a well-defined error interface (similar to SubmissionErrors, but generic)
- throw on error

Reply:
I don't have a good idea for this at the moment, but coupling edit-dialog with final-form is not the way to go.

Reply:
How can I proceed with this issue? How could such an error interface look?

And is a dedicated error interface even necessary in the EditDialog case?
Because the EditDialog doesn't have to know anything about the error. It only needs to know if an error occured (=> do not close) or if no error occured (=> do close).
The error handling is (or should be) done by the child component (e.g. FinalForm)

For reference: `SubmissionErrors` is just an `AnyObject` (see https://github.com/final-form/final-form/blob/master/src/index.d.ts#L8).
So we could also simply use an `AnyObject` in its place.

Reply:
Maybe a general error format for the whole comet admin library would be great. So you always know what you receive if something fails. Is that what you meant @nsams?

e.g. something like

```
interface CometAdminError {
     message?: text;
     ...
     error: AnyObject; (This contains the original error)
}
```